plugins/modules/inventory_intent.py

```diff
@@ -713,59 +985,232 @@ def get_diff_merged(self, config):
+                                    self.msg = "Interface Updation get failed because of {0}".format(execution_details.get("failureReason"))
+                                    self.status = "failed"
+                                    break
+
+                    except Exception as e:
+                        error_message = "Error while Updating Interface Details in Cisco DNA Center - {0}".format(str(e))
+                        self.status = "success"
+                        self.result['changed'] = False
+                        log_msg = "Port actions are only supported on user facing/access ports as it's not allowed or No Updation required"
+                        self.result['log'] = log_msg
+
+                return self
+
+            else:
+                try:
+                    self.mandatory_parameter().check_return_status()
+                    response = self.dnac._exec(
+                        family="devices",
+                        function='sync_devices',
+                        op_modifies=True,
+                        params=self.want.get("device_params"),
+                    )
+
+                    log(str(response))
+
+                    if response and isinstance(response, dict):
+                        task_id = response.get('response').get('taskId')
+                        while True:
+                            execution_details = self.get_task_details(task_id)
+
+                            if execution_details.get("endTime"):
+                                self.status = "success"
+                                self.result['changed'] = True
+                                self.result['response'] = execution_details
+                                break
+                            elif execution_details.get("isError") and execution_details.get("failureReason"):
+                                self.msg = "Device Updation get failed because of {0}".format(execution_details.get("failureReason"))
+                                self.status = "failed"
+                                break
+
+                        log("Device Updated Successfully")
+                        log("Updated devices are :" + str(device_to_update))
+                        msg = "Device " + str(device_to_update) + " updated Successfully !!"
+                        self.result['log'].append(msg)
+
+                except Exception as e:
+                    error_message = "Error while Updating device in Cisco DNA Center - {0}".format(str(e))
+                    self.log(error_message)
+                    raise Exception(error_message)
+
+                msg = "Devices {0} present in Cisco DNA Center and updated successfully".format(config['ip_address'])
+                self.log(msg)
+                self.result['log'].append(msg)
+                self.status = "success"
 
-            log(str(response))
-            device_added = True
+            return self
+
+        else:
```

Comment:
this else also not required. we can restructure the code

Reply:
Make the required changes.